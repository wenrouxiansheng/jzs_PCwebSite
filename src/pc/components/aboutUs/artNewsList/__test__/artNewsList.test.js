import React from "react";
import renderer from "react-test-renderer";
import { shallow, configure, render } from "enzyme";
import ArtNewsList from "../index";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("PC:artNewsList", () => {
  it("渲染测试", () => {
    const component = renderer.create(<ArtNewsList />);
    let tree = component.toJSON();

    //渲染测试
    expect(tree).toMatchSnapshot();
  });

  it("分页事件测试", () => {
    const num = 3;
    const component = shallow(<ArtNewsList />);
    // const a = component.find("Pagination").simulate("change", num).prop('pageSize')
    // component.find("Pagination").simulate("change", a)

    expect(component.find("Pagination").simulate("change", num)).toBe(3);
  });
});
