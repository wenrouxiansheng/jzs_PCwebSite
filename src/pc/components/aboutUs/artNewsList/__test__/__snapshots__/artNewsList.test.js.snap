// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PC:artNewsList 渲染测试 1`] = `
<div
  className="artNewsList"
  style={Object {}}
>
  <div
    className="news"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      className="container"
    >
      <h1>
        最小选手还没话筒高，桔子树金秋朗诵节开始啦！！
        <img
          alt=""
          src="test-file-stub"
        />
      </h1>
      <p>
        噜啦啦噜啦啦噜啦噜啦。。。窗外的天气，就像是你多变的表情，下雨了，雨陪我哭泣，听雨的声音，一滴滴清晰，屋内的湿气香椿储存爱情的记忆，真希望雨能下不停，雨爱的秘密，能一直延续，我相信我将能看见彩虹的美丽
      </p>
      <time>
        2020-12-17
      </time>
    </div>
  </div>
  <div
    className="news"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      className="container"
    >
      <h1>
        最小选手还没话筒高，桔子树金秋朗诵节开始啦！！
        <img
          alt=""
          src="test-file-stub"
        />
      </h1>
      <p>
        噜啦啦噜啦啦噜啦噜啦。。。窗外的天气，就像是你多变的表情，下雨了，雨陪我哭泣，听雨的声音，一滴滴清晰，屋内的湿气香椿储存爱情的记忆，真希望雨能下不停，雨爱的秘密，能一直延续，我相信我将能看见彩虹的美丽
      </p>
      <time>
        2020-12-17
      </time>
    </div>
  </div>
  <div
    className="news"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      className="container"
    >
      <h1>
        最小选手还没话筒高，桔子树金秋朗诵节开始啦！！
        <img
          alt=""
          src="test-file-stub"
        />
      </h1>
      <p>
        噜啦啦噜啦啦噜啦噜啦。。。窗外的天气，就像是你多变的表情，下雨了，雨陪我哭泣，听雨的声音，一滴滴清晰，屋内的湿气香椿储存爱情的记忆，真希望雨能下不停，雨爱的秘密，能一直延续，我相信我将能看见彩虹的美丽
      </p>
      <time>
        2020-12-17
      </time>
    </div>
  </div>
  <div
    className="news"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      className="container"
    >
      <h1>
        最小选手还没话筒高，桔子树金秋朗诵节开始啦！！
        <img
          alt=""
          src="test-file-stub"
        />
      </h1>
      <p>
        噜啦啦噜啦啦噜啦噜啦。。。窗外的天气，就像是你多变的表情，下雨了，雨陪我哭泣，听雨的声音，一滴滴清晰，屋内的湿气香椿储存爱情的记忆，真希望雨能下不停，雨爱的秘密，能一直延续，我相信我将能看见彩虹的美丽
      </p>
      <time>
        2020-12-17
      </time>
    </div>
  </div>
  <div
    className="news"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      className="container"
    >
      <h1>
        最小选手还没话筒高，桔子树金秋朗诵节开始啦！！
        <img
          alt=""
          src="test-file-stub"
        />
      </h1>
      <p>
        噜啦啦噜啦啦噜啦噜啦。。。窗外的天气，就像是你多变的表情，下雨了，雨陪我哭泣，听雨的声音，一滴滴清晰，屋内的湿气香椿储存爱情的记忆，真希望雨能下不停，雨爱的秘密，能一直延续，我相信我将能看见彩虹的美丽
      </p>
      <time>
        2020-12-17
      </time>
    </div>
  </div>
  <div
    className="news"
  >
    <img
      alt=""
      src="test-file-stub"
    />
    <div
      className="container"
    >
      <h1>
        最小选手还没话筒高，桔子树金秋朗诵节开始啦！！
        <img
          alt=""
          src="test-file-stub"
        />
      </h1>
      <p>
        噜啦啦噜啦啦噜啦噜啦。。。窗外的天气，就像是你多变的表情，下雨了，雨陪我哭泣，听雨的声音，一滴滴清晰，屋内的湿气香椿储存爱情的记忆，真希望雨能下不停，雨爱的秘密，能一直延续，我相信我将能看见彩虹的美丽
      </p>
      <time>
        2020-12-17
      </time>
    </div>
  </div>
  <div
    className="pagination"
  >
    <ul
      className="ant-pagination"
      style={Object {}}
      unselectable="unselectable"
    >
      <li
        aria-disabled={true}
        className="ant-pagination-prev ant-pagination-disabled"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={null}
        title="Previous Page"
      >
        <button
          className="ant-pagination-item-link"
          disabled={true}
          tabIndex={-1}
          type="button"
        >
          <span
            aria-label="left"
            className="anticon anticon-left"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="left"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M724 218.3V141c0-6.7-7.7-10.4-12.9-6.3L260.3 486.8a31.86 31.86 0 000 50.3l450.8 352.1c5.3 4.1 12.9.4 12.9-6.3v-77.3c0-4.9-2.3-9.6-6.1-12.6l-360-281 360-281.1c3.8-3 6.1-7.7 6.1-12.6z"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-1 ant-pagination-item-active"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={1}
      >
        <a
          rel="nofollow"
        >
          1
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-2"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={2}
      >
        <a
          rel="nofollow"
        >
          2
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-3"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={3}
      >
        <a
          rel="nofollow"
        >
          3
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-4"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={4}
      >
        <a
          rel="nofollow"
        >
          4
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-5 ant-pagination-item-before-jump-next"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={5}
      >
        <a
          rel="nofollow"
        >
          5
        </a>
      </li>
      <li
        className="ant-pagination-jump-next ant-pagination-jump-next-custom-icon"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title="Next 5 Pages"
      >
        <a
          className="ant-pagination-item-link"
        >
          <div
            className="ant-pagination-item-container"
          >
            <span
              aria-label="double-right"
              className="anticon anticon-double-right ant-pagination-item-link-icon"
              role="img"
            >
              <svg
                aria-hidden="true"
                data-icon="double-right"
                fill="currentColor"
                focusable="false"
                height="1em"
                viewBox="64 64 896 896"
                width="1em"
              >
                <path
                  d="M533.2 492.3L277.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H188c-6.7 0-10.4 7.7-6.3 12.9L447.1 512 181.7 851.1A7.98 7.98 0 00188 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5zm304 0L581.9 166.1c-3-3.9-7.7-6.1-12.6-6.1H492c-6.7 0-10.4 7.7-6.3 12.9L751.1 512 485.7 851.1A7.98 7.98 0 00492 864h77.3c4.9 0 9.6-2.3 12.6-6.1l255.3-326.1c9.1-11.7 9.1-27.9 0-39.5z"
                />
              </svg>
            </span>
            <span
              className="ant-pagination-item-ellipsis"
            >
              •••
            </span>
          </div>
        </a>
      </li>
      <li
        className="ant-pagination-item ant-pagination-item-34"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex="0"
        title={34}
      >
        <a
          rel="nofollow"
        >
          34
        </a>
      </li>
      <li
        aria-disabled={false}
        className="ant-pagination-next"
        onClick={[Function]}
        onKeyPress={[Function]}
        tabIndex={0}
        title="Next Page"
      >
        <button
          className="ant-pagination-item-link"
          disabled={false}
          tabIndex={-1}
          type="button"
        >
          <span
            aria-label="right"
            className="anticon anticon-right"
            role="img"
          >
            <svg
              aria-hidden="true"
              data-icon="right"
              fill="currentColor"
              focusable="false"
              height="1em"
              viewBox="64 64 896 896"
              width="1em"
            >
              <path
                d="M765.7 486.8L314.9 134.7A7.97 7.97 0 00302 141v77.3c0 4.9 2.3 9.6 6.1 12.6l360 281.1-360 281.1c-3.9 3-6.1 7.7-6.1 12.6V883c0 6.7 7.7 10.4 12.9 6.3l450.8-352.1a31.96 31.96 0 000-50.4z"
              />
            </svg>
          </span>
        </button>
      </li>
      <li
        className="ant-pagination-options"
      >
        <div
          className="ant-pagination-options-quick-jumper"
        >
          Go to
          <input
            onBlur={[Function]}
            onChange={[Function]}
            onKeyUp={[Function]}
            type="text"
            value=""
          />
          
        </div>
      </li>
    </ul>
  </div>
</div>
`;
